### DISCUSSION TITLE
Add support for benchmarking time series databases

### DISCUSSION THREAD

**Discussion by valyala** [TPC-Council/HammerDB#128] [Q&A] [answered]
It would be great if HammerDB could support benchmarking for popular TSDBs such as VictoriaMetrics, TimescaleDB, M3DB, Cortex, InfluxDB, etc.

**Comment by sm-shaw** [comment, marked as answer]
It is possible for anyone to add a database to HammerDB and there is a guide here https://www.hammerdb.com/blog/uncategorized/how-to-add-your-database-to-hammerdb-pt1-opening-an-issue/.  There is not currently plans for the core team to add additional databases in the very near future. As background for adding a database one of the first references is the db-engines website, popularity is a key driver of use and it is no coincidence that HammerDB supports the top 5 relational databases.  The higher on this popularity ranking the more likely it will be a candidate for inclusion.  Other factors include the databases and clients running on both Windows and Linux platforms for testing as HammerDB aims to support both equally and the features to build both TPC-C and TPC-H schemas with support for features typcially such as stored procedures for C and parallel query or column store for H.  Currently SAP HANA is highest up the list for inclusion (although the database does not support Windows as an OS).  Previously other databases namely particular Trafodion and Redis were added, however Trafodion support was deprecated at the last release and Redis will be at the next. In particular for these databases performance was not good. For example Redis is a single-threaded database and therefore an entire instance performs the same as 1 virtual user on one of the other databases it also does not support stored procedures and driving a workload entirely from the client side is not scalable for database testing as it involves too many round trips working the connectivity rather than the database.  Probsably not coincidentally these databases did not see as much use as the top 5 and therefore the focus for HammerDB support is on the most popular relational databases.  The best approach if interested in adding other databases is to start with the blog posts linked above.

**Comment by sm-shaw** [comment]
closed as no further follow up
